src/darwin/Framework/CHIP/MTRDeviceControllerDelegateBridge.mm

```diff
@@ -73,12 +80,21 @@
                 [strongDelegate controller:strongController statusUpdate:commissioningStatus];
             });
         }
+
+        // If PASE session setup fails and the client implements the delegate that accepts metrics, invoke the delegate
+        // to mark end of commissioning request.
+        // Since OnPairingComplete(failure_code) might not be invoked in all cases, use this opportunity to inform of failed commissioning
+        // and default the error to timeout since that is best guess in this layer.
+        if (status == chip::Controller::DevicePairingDelegate::Status::SecurePairingFailed && [strongDelegate respondsToSelector:@selector(controller:commissioningComplete:nodeID:metrics:)]) {
+            OnCommissioningComplete(mDeviceNodeId, CHIP_ERROR_TIMEOUT);
```

Comment:
Well, one option is to set this only if we successfully kick off an async pairing operation, right?  As in, don't set this until the call into the C++ returns success.

Reply:
But in that case, the callback above won't be invoked and clobbering the NodeId for the latest one should be fine. Right?

Reply:
So the situation is:

1. Start PASE with node id 1.
2. Try to start PASE with node id 2, that fails to start (because the core SDK can only do one thing at a time for now), but sets mDeviceNodeId to 2.
3. Now PASE to node id 1 fails, reports OnCommissioningComplete for node id 2.

Reply:
Ah I see, I can fix it in next iteration